File: src/ch2/jisyo.py
import sys

dicfile = "./data/ejdict-hand-utf8.txt"

if len(sys.argv) != 2:
    print("[USAGE] jisyo.py word")
    quit()

word = sys.argv[1]

with open(dicfile, "rt", encoding="utf-8") as fp:
    while True:
        line = fp.readline()
        if not line: break
        if word in line:
            print(line.strip())

File: src/ch2/jisyo.rs
use std::fs::File;
use std::io::{BufRead, BufReader};

fn main() {
    let dicfile = "./data/ejdict-hand-utf8.txt";

    let args: Vec<String> = std::env::args().collect();

    if args.len() != 2 {
        println!("[USAGE] jisyo word");
        return;
    }

    let word = &args[1];

    let fp = File::open(dicfile).unwrap();
    let reader = BufReader::new(fp);
    for line in reader.lines() {
        let line = line.unwrap();
        if line.find(word) == None { continue; }
        println!("{}", line);
    }
}